app/src/main/java/com/github/steroidteam/todolist/user/User.java

```diff
@@ -0,0 +1,54 @@
+package com.github.steroidteam.todolist.user;
+
+import com.google.android.gms.tasks.OnCompleteListener;
+
+public interface User {
+    /* Trivial method to have here */
+    boolean isLoggedIn();
```

Comment:
I do not understand under which circumstances could `user.isLoggedIn()` be `false`. If the user has not authenticated, should not `user` be `null`?

That is what I would expect, especially considering that actual authentication takes place upon instantiation of the class (if I understood correctly).

Reply:
Indeed, according to Firebase's documentation the returned user object should be non-null once a user is authenticated. However the authenticated user has a limited lifetime (expired authentication token, logged out etc)